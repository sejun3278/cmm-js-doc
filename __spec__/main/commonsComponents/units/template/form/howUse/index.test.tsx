import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { RecoilRoot } from "recoil";

import _HowUseForm from "src/main/commonsComponents/units/template/form/howUse";
import { useRouter } from "next/router";

// 가짜 router 만들기
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
const push = jest.fn();
(useRouter as jest.Mock).mockImplementation(() => ({
  push,
}));

export default describe("HowUse Template Page", () => {
  // 스냅샷
  test("HowUse Template Page - Snapshot", () => {
    const { container } = render(
      <RecoilRoot>
        <_HowUseForm />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });
});
