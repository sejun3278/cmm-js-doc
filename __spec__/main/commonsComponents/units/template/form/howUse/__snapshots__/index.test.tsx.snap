// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowUse Template Page HowUse Template Page - Snapshot 1`] = `
<div>
  <div
    class="_subTitleTemplate_ css-k59gj9"
  >
    <div>
      <h2
        class="_title_ _subTitle_"
      />
    </div>
    <div
      class="_subTitleLine_ css-18klhw4"
    />
    <div
      class="css-1m3zms3"
    >
      <p
        class="_p_ css-pzlb5d"
      >
        기본값으로 사용할 수 있는 예시 코드입니다.
      </p>
      <div
        class="_copy_ css-que9y8"
      >
        <div
          class="css-eppsv9"
        >
          <pre
            class="css-1gdhu95"
          >
            <code>
              <span
                class="purple"
              >
                import
              </span>
              <span
                class="yellow"
              >
                 { 
              </span>
              <span
                class="skyblue"
              >
                _
              </span>
              <span
                class="yellow"
              >
                 } 
              </span>
              <span
                class="purple"
              >
                from
              </span>
              <span
                class="lightOrange"
              >
                 "MCM-js"
              </span>
              <span
                class="sub"
              >
                ;
              </span>
            </code>
          </pre>
        </div>
        <button
          class="css-12vdm2a"
        >
          <img
            class="_image_ _copyIcon_ css-msby2z"
            src="/images/commons/icons/copy.png"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
